pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -90,6 +89,7 @@ let
 
   inherit (callPackage ./hooks {
     inherit dotnet-sdk dotnet-test-sdk disabledTests nuget-source dotnet-runtime runtimeDeps buildType;
+    runtimeId = dotnetCorePackages.systemToDotnetRid stdenvNoCC.targetPlatform.system;
```

Comment:
I would prefer for this value to be exposed in the top 'buildDotnetModule' as an argument, with the default value set as above. Then nixpkgs pkgs would use the default value, but you also wouldn't break some workflows that use this function outside nixpkgs (like ie. cross-building for windows by setting the runtime id)

Reply:
I made this change:

```diff
commit 18847999cad9611178d086b8d53877877a2cbd2f
Author: David McFarland <corngood@gmail.com>
Date:   Thu Dec 8 12:40:10 2022 -0400

    fixup! build-dotnet-module: restore for current runtime by default

diff --git a/pkgs/build-support/dotnet/build-dotnet-module/default.nix b/pkgs/build-support/dotnet/build-dotnet-module/default.nix
index e297bd3bbcd..273dbd35ff7 100644
--- a/pkgs/build-support/dotnet/build-dotnet-module/default.nix
+++ b/pkgs/build-support/dotnet/build-dotnet-module/default.nix
@@ -59,6 +59,9 @@
   # Libraries that need to be available at runtime should be passed through this.
   # These get wrapped into `LD_LIBRARY_PATH`.
 , runtimeDeps ? [ ]
+  # The dotnet runtime ID. If null, fetch-deps will gather dependencies for all
+  # platforms in meta.platforms which are supported by the sdk.
+, runtimeId ? null
 
   # Tests to disable. This gets passed to `dotnet test --filter "FullyQualifiedName!={}"`, to ensure compatibility with all frameworks.
   # See https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test#filter-option-details for more details.
@@ -90,7 +93,10 @@ let
 
   inherit (callPackage ./hooks {
     inherit dotnet-sdk dotnet-test-sdk disabledTests nuget-source dotnet-runtime runtimeDeps buildType;
-    runtimeId = dotnetCorePackages.systemToDotnetRid stdenvNoCC.targetPlatform.system;
+    runtimeId =
+      if runtimeId != null
+      then [ runtimeId ]
+      else dotnetCorePackages.systemToDotnetRid stdenvNoCC.targetPlatform.system;
   }) dotnetConfigureHook dotnetBuildHook dotnetCheckHook dotnetInstallHook dotnetFixupHook;
 
   localDeps =
@@ -158,7 +164,10 @@ stdenvNoCC.mkDerivation (args // {
     fetch-deps =
       let
         flags = dotnetFlags ++ dotnetRestoreFlags;
-        runtimeIds = map (system: dotnetCorePackages.systemToDotnetRid system) platforms;
+        runtimeIds =
+          if runtimeId != null
+          then [ runtimeId ]
+          else map (system: dotnetCorePackages.systemToDotnetRid system) platforms;
       in
       writeShellScript "fetch-${pname}-deps" ''
         set -euo pipefail
```

I was able to fetch-deps and build a dotnet package on linux using `runtimeId = "win-x64"`.

One potential problem with this is that you can't specify multiple RIDs to restore in fetch-deps.  Is that a problem for you?

If you have any examples of how you use this, I could make some tests.

Reply:
Yeah, this fixes my use case and should be enough.

I don't see the problem with the `fetch-deps` logic, it should always restore for all platforms specified. Maybe include windows rids in `systemToDotnetRid`? I know nixpkgs has windows targets specified.

Reply:
I added RIDs for the two windows configs in doubles.nix.